docs/0193.md

```diff
@@ -0,0 +1,104 @@
+# Rules for AEP-193: Errors
+
+[aep-193]: https://aep.dev/193
+
+## aep-193-error-response-schema
+
+**Rule**: Error response schema should contain all required fields of the
+correct type.
+
+This rule enforces that all 4xx and 5xx responses have valid schema according
+to AEP-193 rule.
+
+### Details
+
+This rule looks at the 4xx and 5xx responses and reports if either one of
+required parameters is missing or if it has incorrect type.
+
+### Examples
+
+**Incorrect** code for this rule:
+
+```yaml
+paths:
+  '/test1':
+    get:
+      description: '404 response schema is missing required fields: type, detail, instance and title'
+      responses:
+        200:
+          content:
+            'application/json':
+              schema:
+                type: 'object'
+                properties:
+                  results:
+                    type: string
+        404:
+          content:
+            'application/json':
+              schema:
+                type: 'object'
+                properties:
+                  status:
+                    type: 'integer'
+  '/test2':
+    get:
+      description: '404 response schema detail property is of incorrect type'
+      404:
+        content:
+          'application/json':
+            schema:
+              type: 'object'
+              properties:
+                status:
+                  type: 'integer'
+                type:
+                  type: 'string',
+                detail:
+                  type: 'integer',
+                instance:
+                  type: 'string',
+                title:
+                  type: 'string',
+```
+
+**Correct** code for this rule:
+
+```yaml
+paths:
+  '/books':
+    get:
+      responses:
+        404:
+          content:
+            'application/json':
+              schema:
+                type: 'object'
```

Comment:
probably should just copy the problem details schema as is: https://datatracker.ietf.org/doc/html/rfc9457#name-json-schema-for-http-proble